it/controllers/benefits/income/IncomeTaxBenefitsAmountControllerISpec.scala

```diff
@@ -0,0 +1,546 @@
+        authoriseAgentOrIndividual(isAgent = false)
+        dropEmploymentDB()
+        userDataStub(userData(fullEmploymentsModel()), nino, taxYearEOY)
+        insertCyaData(employmentUserData(hasPriorBenefits = true, cyaModel(
+          Some(benefits(fullIncomeTaxOrIncurredCostsModel)))), userRequest)
+        urlGet(pageUrl(taxYear), follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)))
+      }
+
+      s"has an SEE OTHER($SEE_OTHER) status" in {
+        result.status shouldBe SEE_OTHER
+        result.header("location") shouldBe Some(appConfig.incomeTaxSubmissionOverviewUrl(taxYear))
+      }
+    }
+
+    "redirect to the check your benefits page when there is no cya data found" which {
+      implicit lazy val result: WSResponse = {
+        authoriseAgentOrIndividual(isAgent = false)
+        dropEmploymentDB()
+        urlGet(pageUrl(taxYearEOY), follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+      }
+
+      s"has an SEE OTHER($SEE_OTHER) status" in {
+        result.status shouldBe SEE_OTHER
+        result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url)
+      }
+    }
+    "redirect to the check your benefits page when the income tax benefits question is set to false" which {
+      implicit lazy val result: WSResponse = {
+        authoriseAgentOrIndividual(isAgent = false)
+        dropEmploymentDB()
+        userDataStub(userData(fullEmploymentsModel()), nino, taxYearEOY)
+        insertCyaData(employmentUserData(hasPriorBenefits = true, cyaModel(
+          Some(benefits(fullIncomeTaxOrIncurredCostsModel.copy(incomeTaxPaidByDirectorQuestion = Some(false), incomeTaxPaidByDirector = None))))), userRequest)
+        urlGet(pageUrl(taxYearEOY), follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)))
+      }
+
+      "has an SEE_OTHER(303) status" in {
+        result.status shouldBe SEE_OTHER
+        result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url)
+      }
+    }
+
+    "redirect to the check your benefits page when the income tax or incurred costs question is set to false" which {
+      implicit lazy val result: WSResponse = {
+        authoriseAgentOrIndividual(isAgent = false)
+        dropEmploymentDB()
+        userDataStub(userData(fullEmploymentsModel()), nino, taxYearEOY)
+        insertCyaData(employmentUserData(hasPriorBenefits = true, cyaModel(
+          Some(benefits(emptyIncomeTaxOrIncurredCostsModel)))), userRequest)
+        urlGet(pageUrl(taxYearEOY), follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)))
+      }
+
+      "has an SEE_OTHER(303) status" in {
+        result.status shouldBe SEE_OTHER
+        result.header("location") shouldBe Some(s"/update-and-submit-income-tax-return/employment-income/$taxYearEOY/check-employment-benefits?employmentId=001")
```

Comment:
can you use the route check instead of the direct url, same as you did on line 290

Reply:
changed to reflect line 290